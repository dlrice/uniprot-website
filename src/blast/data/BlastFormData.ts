export type BlastFormValue = {
  fieldName: string;
  selected?: string | undefined;
  selectedLabel?: string | undefined;
  type: BlastFieldTypes;
  values?: { label?: string; value?: string | number }[];
};

export enum BlastFields {
  program = 'Program',
  stype = 'Sequence type',
  email = 'Email',
  sequence = 'Sequence',
  taxon = 'Taxon',
  targetDb = 'Target database',
  threshold = 'E-Threshold',
  matrix = 'Matrix',
  filter = 'Filter',
  gapped = 'Gapped',
  hits = 'Hits',
}

export enum BlastFieldTypes {
  textarea,
  select,
  autocomplete,
}

export type BlastFormValues = { [x in BlastFields]: BlastFormValue };

export default {
  [BlastFields.program]: {
    fieldName: 'program',
    selected: 'blastp',
  },
  [BlastFields.stype]: {
    fieldName: 'stype',
    selected: 'protein',
  },
  [BlastFields.email]: {
    fieldName: 'email',
    selected: 'uuw_dev@uniprot.org',
  },
  [BlastFields.sequence]: {
    fieldName: 'sequence',
    type: BlastFieldTypes.textarea,
  },
  [BlastFields.targetDb]: {
    fieldName: 'database',
    type: BlastFieldTypes.select,
    selected: 'uniprotkb_refprotswissprot',
    values: [
      {
        value: 'uniprotkb_refprotswissprot',
        label: 'UniProtKb',
      },
    ],
  },
  [BlastFields.taxon]: {
    fieldName: 'taxids',
    type: BlastFieldTypes.autocomplete,
  },
  [BlastFields.threshold]: {
    fieldName: 'threshold',
    type: BlastFieldTypes.select,
    selected: '10',
    values: [
      { value: '0.0001' },
      { value: '0.001' },
      { value: '0.01' },
      { value: '0.1' },
      { value: '1' },
      { value: '10' },
      { value: '100' },
      { value: '1000' },
    ],
  },
  [BlastFields.matrix]: {
    fieldName: 'matrix',
    type: BlastFieldTypes.select,
    selected: null,
    values: [
      {
        value: '',
        label: 'Auto',
      },
      { value: '45' },
      { value: '62' },
      { value: '80' },
      { value: '70' },
      { value: '30' },
    ],
  },
  [BlastFields.filter]: {
    fieldName: 'filter',
    type: BlastFieldTypes.select,
    selected: 'F',
    values: [
      { value: 'F', label: 'None' },
      { valueNone: 'T', label: 'Filter low complexity regions' },
      { value: 'mask', label: 'Mask lookup table only' },
    ],
  },
  [BlastFields.gapped]: {
    fieldName: 'gapped',
    type: BlastFieldTypes.select,
    selected: 'true',
    values: [
      { value: 'true', label: 'yes' },
      { value: 'false', label: 'no' },
    ],
  },
  [BlastFields.hits]: {
    fieldName: 'hits',
    type: BlastFieldTypes.select,
    selected: '250',
    values: [
      { value: '50' },
      { value: '100' },
      { value: '250' },
      { value: '500' },
      { value: '750' },
      { value: '1000' },
    ],
  },
};
