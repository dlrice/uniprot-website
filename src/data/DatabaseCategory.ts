enum databaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

export default databaseCategory;
