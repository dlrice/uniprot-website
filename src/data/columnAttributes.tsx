const small = 200; //10rem;
const medium = 400; //15rem;
const large = 600; //20rem;

export type columnAttributes = {
  [columnName: string]: {
    width: number;
    label: string;
  };
};

const columnAttributes: columnAttributes = {
  accession: {
    width: small,
    label: 'Entry',
  },
  id: {
    width: small,
    label: 'Entry Name',
  },
  protein_name: {
    width: large,
    label: 'Protein names',
  },
  gene_names: {
    width: medium,
    label: 'Gene Names',
  },
  organism: {
    width: large,
    label: 'Organism',
  },
};

export default columnAttributes;
