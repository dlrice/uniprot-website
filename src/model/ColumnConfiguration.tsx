import React from 'react';
import SimpleView from '../view/uniprotkb/components/SimpleView';
import { Organism } from '../view/uniprotkb/components/OrganismView';
import {
  ProteinNamesData,
  GeneNamesData,
  OrganismData,
} from './uniprotkb/sections/NamesAndTaxonomyConverter';

const WIDTH = {
  small: 200,
  medium: 400,
  large: 600,
};

const ColumnConfiguration: {
  [index: string]: {
    width: number;
    label: string;
    render: (data: any) => JSX.Element | undefined;
  };
} = {
  accession: {
    width: WIDTH.small,
    label: 'Entry',
    render: (data: { primaryAccession: string; entryType: string }) => (
      <SimpleView
        termValue={`${data.primaryAccession} (${data.entryType})`}
        linkTo={`/uniprotkb/${data.primaryAccession}`}
      />
    ),
  },
  id: {
    width: WIDTH.small,
    label: 'Entry Name',
    render: (data: { uniProtId: string }) => (
      <SimpleView termValue={data.uniProtId} />
    ),
  },
  protein_name: {
    width: WIDTH.large,
    label: 'Protein names',
    render: (data: ProteinNamesData) => <span />,
    // TODO TEMPORARY!!!
    // <ProteinNames data={data} />,
  },
  gene_names: {
    width: WIDTH.medium,
    label: 'Gene Names',
    render: (data: GeneNamesData) => <span />,
    // TODO TEMPORARY!!!
    // <ProteinNames data={data} />,
  },
  organism: {
    width: WIDTH.large,
    label: 'Organism',
    render: (data: OrganismData) => <span />,
    // TODO TEMPORARY!!!
    // (data: OrganismData) => <Organism data={data} />,
  },
};

export default ColumnConfiguration;
