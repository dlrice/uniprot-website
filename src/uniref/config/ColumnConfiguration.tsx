/* eslint-disable camelcase */
import React from 'react';
import { Link } from 'react-router-dom';
import { UniRefUIModel } from '../adapters/uniRefConverter';

export enum UniRefColumn {
  id = 'id',
  name = 'name',
  commonTaxon = 'common_taxon',
  commonTaxonid = 'common_taxonid',
  organismId = 'organism_id',
  organism = 'organism',
  identity = 'identity',
  length = 'length',
  sequence = 'sequence',
  types = 'types',
  members = 'members',
  count = 'count',
  created = 'created',
}

export const ColumnConfiguration = new Map<
  UniRefColumn,
  {
    label: string;
    render: (data: UniRefUIModel) => JSX.Element | string | number | undefined;
  }
>();

ColumnConfiguration.set(UniRefColumn.id, {
  label: 'Cluster ID',
  render: ({ id }) => {
    return id;
  },
});

ColumnConfiguration.set(UniRefColumn.name, {
  label: 'Cluster name',
  render: ({ name }) => {
    return name;
  },
});

ColumnConfiguration.set(UniRefColumn.commonTaxon, {
  label: 'Common taxon',
  render: ({ commonTaxon }) => {
    return commonTaxon;
  },
});

ColumnConfiguration.set(UniRefColumn.commonTaxonid, {
  label: 'Common taxon ID',
  render: ({ commonTaxonId }) => {
    return <Link to={`/taxonomy/${commonTaxonId}`}>{commonTaxonId}</Link>;
  },
});

ColumnConfiguration.set(UniRefColumn.organismId, {
  label: 'Organism IDs',
  render: ({ members }) => {
    return members
      ?.map((member) => (
        <Link to={`/taxonomy/${member.organismTaxId}`}>
          {member.organismTaxId}
        </Link>
      ))
      .join(', ');
  },
});

ColumnConfiguration.set(UniRefColumn.organism, {
  label: 'Organisms',
  render: ({ members }) => {
    return members?.map((member) => member.organismName).join(', ');
  },
});

// TODO where is this??
// ColumnConfiguration.set(UniRefColumn.identity, {
//   label: 'Identity',
//   render: (data) => {
//     return <></>;
//   },
// });

ColumnConfiguration.set(UniRefColumn.length, {
  label: 'Length',
  render: ({ Sequence }) => Sequence.sequence.length,
});

ColumnConfiguration.set(UniRefColumn.sequence, {
  label: 'Reference sequence',
  render: (data) => {
    return data.Sequence.sequence.value;
  },
});

ColumnConfiguration.set(UniRefColumn.types, {
  label: 'Types',
  render: ({ entryType }) => {
    // Note: not sure this is the right one
    return entryType;
  },
});

ColumnConfiguration.set(UniRefColumn.members, {
  label: 'Members',
  render: ({ members }) => {
    return members
      ?.map((member) =>
        member.accessions.map((accession) => (
          <Link to={`/uniprotkb/${accession}`}>{accession}</Link>
        ))
      )
      .join(', ');
  },
});

ColumnConfiguration.set(UniRefColumn.count, {
  label: 'Size',
  render: ({ memberCount }) => {
    return memberCount;
  },
});

ColumnConfiguration.set(UniRefColumn.created, {
  label: 'Date of creation',
  render: ({ updated }) => {
    // Note this is actually an update , not creation
    return updated;
  },
});

export default ColumnConfiguration;
