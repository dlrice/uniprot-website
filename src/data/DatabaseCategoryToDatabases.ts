import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string[]>([
  [
    DatabaseCategory.SEQUENCE,
    ['EMBL/GenBank/DDBJ', 'CCDS', 'PIR', 'RefSeq', 'UniGene'],
  ],
  [DatabaseCategory.STRUCTURE, ['PDB', 'SMR', 'DisProt', 'ProteinModelPortal']],
  [
    DatabaseCategory.INTERACTION,
    ['BioGrid', 'DIP', 'IntAct', 'MINT', 'STRING'],
  ],
  [
    DatabaseCategory.CHEMISTRY,
    ['BindingDB', 'ChEMBL', 'DrugBank', 'GuidetoPHARMACOLOGY'],
  ],
  [
    DatabaseCategory.FAMILY,
    [
      'Allergome',
      'CAZy',
      'MEROPS',
      'MoonProt',
      'mycoCLAP',
      'PeroxiBase',
      'PptaseDB',
      'REBASE',
      'TCDB',
    ],
  ],
  [DatabaseCategory.PTM, ['DEPOD', 'PhosphoSite', 'PhosSite', 'UniCarbKB']],
  [DatabaseCategory.POLYMORPHISM, ['dbSNP', 'DMDM']],
  [
    DatabaseCategory.GEL,
    [
      'SWISS-2DPAGE',
      'COMPLUYEAST-2DPAGE',
      'DOSAC-COBS-2DPAGE',
      'OGP',
      'REPRODUCTION-2DPAGE',
      'UCD-2DPAGE',
      'World-2DPAGE',
    ],
  ],
  [
    DatabaseCategory.PROTEOMIC,
    ['MaxQB', 'PaxDb', 'PeptideAtlas', 'PRIDE', 'ProMEX'],
  ],
  [DatabaseCategory.PROTOCOL, ['DNASU']],
  [
    DatabaseCategory.GENOME,
    [
      'Ensembl and Ensembl Genomes',
      'Ensembl',
      'EnsemblBacteria',
      'EnsemblFungi',
      'EnsemblMetazoa',
      'EnsemblPlants',
      'EnsemblProtists',
      'GeneCards',
      'GeneID',
      'KEGG',
      'PATRIC',
      'UCSC',
      'VectorBase',
    ],
  ],
  [
    DatabaseCategory.ORGANISM,
    [
      'ArachnoServer',
      'CGD',
      'ConoServer',
      'CTD',
      'dictyBase',
      'EchoBASE',
      'EcoGene',
      'euHCVdb',
      'EuPathDB',
      'FlyBase',
      'GeneReviews',
      'Gramene',
      'H-InvDB',
      'HGNC',
      'HPA',
      'LegioList',
      'Leproma',
      'MaizeGDB',
      'MIM',
      'MGI',
      'neXtProt',
      'Orphanet',
      'PharmGKB',
      'PomBase',
      'PseudoCAP',
      'RGD',
      'SGD',
      'TAIR',
      'TubercuList',
      'WormBase',
      'Xenbase',
      'ZFIN',
    ],
  ],
  [
    DatabaseCategory.PHYLOGENOMIC,
    [
      'eggNOG',
      'GeneTree',
      'HOGENOM',
      'HOVERGEN',
      'InParanoid',
      'KO',
      'OMA',
      'OrthoDB',
      'PhylomeDB',
      'TreeFam',
    ],
  ],
  [
    DatabaseCategory.PATHWAY,
    ['BioCyc', 'BRENDA', 'Reactome', 'SABIO-RK', 'SignaLink', 'UniPathway'],
  ],
  [DatabaseCategory.EXPRESSION, ['Bgee', 'ExpressionAtlas']],
  [
    DatabaseCategory.DOMAIN,
    [
      'HAMAP',
      'InterPro',
      'Gene3D',
      'PANTHER',
      'Pfam',
      'PIRSF',
      'PRINTS',
      'ProDom',
      'SMART',
      'SUPFAM',
      'TIGRFAMs',
      'PROSITE',
    ],
  ],
  [
    DatabaseCategory.OTHER,
    [
      'ChiTaRS',
      'EvolutionaryTrace',
      'GenomeRNAi',
      'GeneWiki',
      'NextBio',
      'PMAP-CutDB',
      'PRO',
    ],
  ],
]);

export default databaseCategoryToDatabases;
