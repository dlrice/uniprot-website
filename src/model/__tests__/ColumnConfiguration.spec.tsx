import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ColumnConfiguration from '../ColumnConfiguration';
import data from '../__mocks__/modelData.json';
import uniProtKbConverter from '../uniprotkb/UniProtkbConverter';
import renderWithRedux from '../../__testHelpers__/renderWithRedux';

describe('ColumnConfiguration component', () => {
  let transformedData;

  beforeAll(() => {
    transformedData = uniProtKbConverter(data);
    window.SVGElement.prototype.getBBox = () => ({
      width: 10,
      height: 10,
    });
  });

  test('should render all columns', () => {
    ColumnConfiguration.forEach(column => {
      const { asFragment } = renderWithRedux(
        <MemoryRouter>{column.render(transformedData)}</MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
