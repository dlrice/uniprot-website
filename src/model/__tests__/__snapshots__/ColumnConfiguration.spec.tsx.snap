// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnConfiguration component should render all columns 1`] = `
<DocumentFragment>
  <a
    href="/uniprotkb/P21802"
  >
    P21802 (Swiss-Prot)
  </a>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 2`] = `
<DocumentFragment>
  <span>
    uniprot id
  </span>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 3`] = `
<DocumentFragment>
  <div>
    <strong>
      some Gene
    </strong>
  </div>
  <div>
    some Syn
  </div>
  <div>
    some orf
  </div>
  <div>
    some locus
  </div>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 4`] = `
<DocumentFragment>
  10 AA
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 5`] = `
<DocumentFragment>
  <div>
    some Gene
  </div>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 6`] = `
<DocumentFragment>
  some locus
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 7`] = `
<DocumentFragment>
  some orf
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 8`] = `
<DocumentFragment>
  some Syn
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 9`] = `
<DocumentFragment>
  <a
    href="/taxonomy/9606"
  >
    scientific name (common name)  (synonyms 1)
  </a>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 10`] = `
<DocumentFragment>
  <a
    href="/taxonomy/9606"
  >
    9606 
  </a>
  <a
    class="external-link"
    href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?lvl=0&id=9606"
    target="_blank"
  >
    NCBI
    <svg
      viewBox="0 0 576 512"
      width="12.5"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 11`] = `
<DocumentFragment>
  <div>
    <strong>
      rec full Name
    </strong>
  </div>
  <div>
    1.2.3.4
  </div>
  <div>
    recommended short name
  </div>
  <div>
    <ul
      class="no-bullet"
    >
      <li>
        a full alt Name (short alt name1) 
      </li>
    </ul>
  </div>
  <div>
    <ul
      class="no-bullet"
    >
      <li>
        containsrec full Name (containsrecommended short name)  
        <strong>
          Alternative names: 
        </strong>
        containsa full alt Name (containsshort alt name1) 
      </li>
    </ul>
  </div>
  <div>
    <ul
      class="no-bullet"
    >
      <li>
        sub full Name
      </li>
    </ul>
  </div>
  <div>
    biotech
  </div>
  <div />
  <div />
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 12`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 13`] = `
<DocumentFragment>
  lineage 1
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 14`] = `
<DocumentFragment>
  <p>
    <a
      href="/taxonomy/9606"
    >
      scientific name (common name)  (synonyms 1)
    </a>
  </p>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 15`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 16`] = `
<DocumentFragment>
  <section
    class="text-block"
  >
    The sequence 
    <a
      class="external-link"
      href="//www.ebi.ac.uk/ena/data/view/sequence"
      target="_blank"
    >
      sequence
      <svg
        viewBox="0 0 576 512"
        width="12.5"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
        />
      </svg>
    </a>
     differs from that shown. Reason: Erroneous initiation Text note
    <span
      class="evidence-tag svg-colour-unreviewed"
      data-testid="evidence-tag-trigger"
      role="button"
      tabindex="0"
    >
      <svg
        height="12"
        viewBox="0 0 400 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 144H256V0l144 144zm-176 32V0H0v512h400V176H224z"
          id="Shape"
        />
      </svg>
      <span
        class="evidence-tag__label"
        title=""
      >
        1 automatic annotation
      </span>
    </span>
    <div
      class="evidence-tag-content "
      data-testid="evidence-tag-content"
    >
      <div>
        <h5>
          Automatic assertion according to rules
        </h5>
        <div>
          PIRNR:PIRNR001361
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 17`] = `
<DocumentFragment>
  Fragment
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 18`] = `
<DocumentFragment>
  1,185 Da
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 19`] = `
<DocumentFragment>
  <section
    class="text-block"
  >
    Molecular mass is 2.1 Da from positions 
    <span>
      1-7
    </span>
    . Determined by LSI. note value 
    <span
      class="evidence-tag svg-colour-unreviewed"
      data-testid="evidence-tag-trigger"
      role="button"
      tabindex="0"
    >
      <svg
        height="12"
        viewBox="0 0 400 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 144H256V0l144 144zm-176 32V0H0v512h400V176H224z"
          id="Shape"
        />
      </svg>
      <span
        class="evidence-tag__label"
        title=""
      >
        1 automatic annotation
      </span>
    </span>
    <div
      class="evidence-tag-content "
      data-testid="evidence-tag-content"
    >
      <div>
        <h5>
          Automatic assertion according to rules
        </h5>
        <div>
          PIRNR:PIRNR001361
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 20`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 21`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 22`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 23`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 24`] = `
<DocumentFragment>
  <section
    class="text-block"
  >
    <div>
      Edited at positions 
      <span>
        rna position 
        <span
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            height="12"
            viewBox="0 0 400 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 144H256V0l144 144zm-176 32V0H0v512h400V176H224z"
              id="Shape"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Automatic assertion according to rules
            </h5>
            <div>
              PIRNR:PIRNR001361
            </div>
          </div>
        </div>
      </span>
    </div>
    <div>
      <span>
        value 
        <span
          class="evidence-tag svg-colour-unreviewed"
          data-testid="evidence-tag-trigger"
          role="button"
          tabindex="0"
        >
          <svg
            height="12"
            viewBox="0 0 400 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 144H256V0l144 144zm-176 32V0H0v512h400V176H224z"
              id="Shape"
            />
          </svg>
          <span
            class="evidence-tag__label"
            title=""
          >
            1 automatic annotation
          </span>
        </span>
        <div
          class="evidence-tag-content "
          data-testid="evidence-tag-content"
        >
          <div>
            <h5>
              Automatic assertion according to rules
            </h5>
            <div>
              PIRNR:PIRNR001361
            </div>
          </div>
        </div>
      </span>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 25`] = `
<DocumentFragment>
  <section
    class="text-block"
  >
    The sequence 
    <a
      class="external-link"
      href="//www.ebi.ac.uk/ena/data/view/sequence"
      target="_blank"
    >
      sequence
      <svg
        viewBox="0 0 576 512"
        width="12.5"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
        />
      </svg>
    </a>
     differs from that shown. Reason: Erroneous initiation Text note
    <span
      class="evidence-tag svg-colour-unreviewed"
      data-testid="evidence-tag-trigger"
      role="button"
      tabindex="0"
    >
      <svg
        height="12"
        viewBox="0 0 400 512"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 144H256V0l144 144zm-176 32V0H0v512h400V176H224z"
          id="Shape"
        />
      </svg>
      <span
        class="evidence-tag__label"
        title=""
      >
        1 automatic annotation
      </span>
    </span>
    <div
      class="evidence-tag-content "
      data-testid="evidence-tag-content"
    >
      <div>
        <h5>
          Automatic assertion according to rules
        </h5>
        <div>
          PIRNR:PIRNR001361
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 26`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 27`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 28`] = `
<DocumentFragment>
  <span>
    5
  </span>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 29`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 30`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 31`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 32`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 33`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 34`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 35`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 36`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 37`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 38`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 39`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 40`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 41`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 42`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 43`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 44`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 45`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 46`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 47`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 48`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 49`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 50`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 51`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 52`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 53`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 54`] = `
<DocumentFragment>
  2015-08-02
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 55`] = `
<DocumentFragment>
  2016-04-24
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 56`] = `
<DocumentFragment>
  2017-01-21
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 57`] = `
<DocumentFragment>
  <span>
    20
  </span>
</DocumentFragment>
`;

exports[`ColumnConfiguration component should render all columns 58`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 59`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 60`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 61`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 62`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 63`] = `<DocumentFragment />`;

exports[`ColumnConfiguration component should render all columns 64`] = `<DocumentFragment />`;
